feat: colored tags

pj-gridx opened the pull request:
## Changes

### Tag color support
- schema/migration adds color with default #f2f4f7  
- tag services accept/persist color  
- CSV/import tags get a random color  

### forms/UI
- tag create/edit use the color picker  
- tag badges support colored backgrounds/dots  
- asset/tag listings, asset overview, and booking stats render colored tags with safe optional color handling  
- color input widened for hex codes  

### booking pipeline
- tag color included in booking queries (constants, pdf helpers, routes)  
- booking service tests updated for color-aware expectations and future-dated fixtures  

### assets/types
- tag color added to asset loader/types  
- list item tags column and booking stats tolerate missing colors  

### housekeeping
- lockfile updated  
- typegen/typecheck cleaned up around optional colors  


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Tags now display with color indicators and styled visual badges throughout the application, including asset views, booking lists, and tag management
  * Users can customize tag colors through a new color input field in tag creation and editing interfaces
  * Tags are automatically assigned a default color to ensure consistent visual representation

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

carlosvirreira commented:
@coderabbit full review

DonKoko reviewed (changes requested):
Thanks for the PR.
Overall it looks good but you have 11k lines changed in `package-lock.json`. Please resolve this as those changes shouldn't be present in the branch. 
Once resolved I will run the branch locally and test it further.

DonKoko commented on app/components/shared/tag.tsx:
This component is used in a lot of places, not only for the tags. You need to double check other uses and make sure its fully backwards compatible and didn't break anything.

DonKoko commented on app/routes/_layout+/tags.$tagId_.edit.tsx:
```diff
@@ -27,6 +28,7 @@ import { zodFieldIsRequired } from "~/utils/zod";
 export const UpdateTagFormSchema = z.object({
   name: z.string().min(3, "Name is required"),
   description: z.string(),
+  color: z.string().regex(/^#/).min(7),
```
This will accept invalid values. Can you please update the regex:
```
color: z.string().regex(/^#[0-9a-fA-F]{6}$/),
```
I think this should be better.

DonKoko commented on app/routes/_layout+/tags.new.tsx:
```diff
@@ -26,6 +28,7 @@ import { zodFieldIsRequired } from "~/utils/zod";
 export const NewTagFormSchema = z.object({
   name: z.string().min(3, "Name is required"),
   description: z.string(),
+  color: z.string().regex(/^#/).min(7),
```
Same as with update tag schema. Improve the regex

DonKoko reviewed (changes requested):
Overall very good PR. There were some conflicts with `main` I had to resolve so make sure to update your local branch before working on it further.
I have added some feedback, nothing major but important to make this complete.

- [x] On the advanced asset index, the styling of the tags is not applied properly. In this case "legendary" has a custom color but its showing as default
<img width="376" height="182" alt="Image" src="https://github.com/user-attachments/assets/a9f2b904-0b79-41d2-a81f-24cbe3732fb6" />

- [ ] On the tags index as well as other views such as the asset page, simple asset index, the default color of tags is broken. It is not accessibility compliant and is very unreadable. Lets make it the default look like it does on the advanced index mode.
<img width="954" height="86" alt="Image" src="https://github.com/user-attachments/assets/c862687a-1a25-4620-93fb-7a120745ad22" />

- [x] On the tags index as well as some other places the tags are showing with a dot. Lets remove that to keep them clean.

pj-gridx commented:
@DonKoko I adjusted the code, thanks for the detailed review.

**Tag visuals refined**
removed the decorative dots on tag badges, matched the tag layout to the badge styling for consistency, and tweaked the default tag color to improve contrast/visibility.

**Color data fixes**:
now pulling the actual tag color from the DB instead of a fallback, using a const select shape for id/name/color to keep typings accurate, and tightened the color regex to allow only 6‑digit hex codes.

DonKoko commented:
Thanks for the great PR @pj-gridx. I just had to make some minor adjustments but overall it was great. I appreciate your input.

pj-gridx commented:
@carlosvirreira @DonKoko thank you both for the great support!
